test/modules/acbs/acbs-authentication.service.test.ts

```diff
@@ -0,0 +1,242 @@
+        });
+
+        it('throws an AcbsAuthenticationFailedException if the IdP returns an empty id_token', async () => {
+            mockSuccessfulCreateSessionRequest();
+            mockSuccessfulGetTokenForSessionRequestReturning({ id_token: '' });
+
+            const getTokenPromise = service.getIdToken();
+
+            await expect(getTokenPromise)
+                .rejects.toBeInstanceOf(AcbsAuthenticationFailedException);
+            await expect(getTokenPromise)
+                .rejects.toThrowError('ID token was not returned by the IdP.');
+            await expect(getTokenPromise)
+                .rejects.toHaveProperty('innerError', undefined);
+        });
+
+        it('throws an AcbsAuthenticationFailedException if the IdP returns an undefined id_token', async () => {
+            mockSuccessfulCreateSessionRequest();
+            mockSuccessfulGetTokenForSessionRequestReturning({ id_token: undefined });
+
+            const getTokenPromise = service.getIdToken();
+
+            await expect(getTokenPromise)
+                .rejects.toBeInstanceOf(AcbsAuthenticationFailedException);
+            await expect(getTokenPromise)
+                .rejects.toThrowError('ID token was not returned by the IdP.');
+            await expect(getTokenPromise)
+                .rejects.toHaveProperty('innerError', undefined);
+        });
+
+        it('throws an AcbsAuthenticationFailedException if the IdP returns a null id_token', async () => {
+            mockSuccessfulCreateSessionRequest();
+            mockSuccessfulGetTokenForSessionRequestReturning({ id_token: null });
+
+            const getTokenPromise = service.getIdToken();
+
+            await expect(getTokenPromise)
+                .rejects.toBeInstanceOf(AcbsAuthenticationFailedException);
+            await expect(getTokenPromise)
+                .rejects.toThrowError('ID token was not returned by the IdP.');
+            await expect(getTokenPromise)
+                .rejects.toHaveProperty('innerError', undefined);
+        });
+    });
+
+    function mockSuccessfulCreateSessionRequest(): void {
+        mockSuccessfulCreateSessionRequestReturningCookies([cookie1, sessionIdCookie, cookie2]);
+    }
+
+    function mockSuccessfulCreateSessionRequestReturningCookies(cookies: string[]): void {
+        const headers = new AxiosHeaders();
+        headers['set-cookie'] = cookies;
+
+        when(httpService.post)
+            .calledWith(...expectedPostSessionsArguments)
+            .mockReturnValueOnce(of({
+                data: '',
+                status: 200,
+                statusText: 'OK',
+                config: undefined,
+                headers: headers,
+            }));
+    }
+
+    function mockSuccessfulGetTokenForSessionRequest(): void {
+        mockSuccessfulGetTokenForSessionRequestReturning({ id_token: idToken });
+    }
+
+    function mockSuccessfulGetTokenForSessionRequestReturning(data: any): void {
```

Comment:
Can we use arrow functions for this?

Reply:
Sure - just to double-check, is that just for test files?

Reply:
For all functions please